src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/IndentBlockFormattingRule.cs

```diff
@@ -246,6 +248,24 @@ private void AddBlockIndentationOperation(List<IndentBlockOperation> list, Synta
             AddIndentBlockOperation(list, bracePair.openBrace.GetNextToken(includeZeroWidth: true), bracePair.closeBrace.GetPreviousToken(includeZeroWidth: true));
         }
 
+        private static void AddBracketIndentationOperation(List<IndentBlockOperation> list, SyntaxNode node)
+        {
+            var bracketPair = node.GetBracketPair();
+
+            if (!bracketPair.IsValidBracePair())
```

Comment:
nit: Consider strengthening the logic in `IsValidBracePair` to check that we either have open-brace/close-brace or open-bracket/close-bracket. Maybe the name of the method should be tweaked? Maybe `IsValidBraceOrBracketPair`?

Reply:
Good idea.. lets see what tests break :)